src/compiler/types.rs

```diff
@@ -1133,10 +1144,51 @@ pub trait LiteralLikeNodeInterface {
 #[ast_type(ancestors = "Expression", interfaces = "LiteralLikeNodeInterface")]
 pub enum LiteralLikeNode {
     StringLiteral(StringLiteral),
+    TemplateLiteralLikeNode(TemplateLiteralLikeNode),
     NumericLiteral(NumericLiteral),
     BigIntLiteral(BigIntLiteral),
 }
 
+#[derive(Debug)]
+#[ast_type(
+    ancestors = "LiteralLikeNode, Expression",
+    interfaces = "LiteralLikeNodeInterface"
+)]
+pub struct TemplateLiteralLikeNode {
+    _literal_like_node: BaseLiteralLikeNode,
+    pub raw_text: Option<String>,
+    pub template_flags: Option<TokenFlags>,
+}
+
+impl TemplateLiteralLikeNode {
+    pub fn new(
+        literal_like_node: BaseLiteralLikeNode,
+        raw_text: Option<String>,
+        template_flags: Option<TokenFlags>,
+    ) -> Self {
+        Self {
+            _literal_like_node: literal_like_node,
+            raw_text,
+            template_flags,
+        }
+    }
+}
+
+impl TemplateLiteralLikeNodeInterface for TemplateLiteralLikeNode {
```

Comment:
This maybe doesn't need to be an interface (vs concrete stuff on `TemplateLiteralLikeNode` since it ended up being a concrete AST type and I don't think there will be other node types implementing this interface